### PR TITLE
FX #9806 Fix crashing when sharing top site on iPad

### PR DESCRIPTION
**Pull request by junbio**
This PR fixes sharing top sites on iPad. (#9806)

### CODE REVIEW

**Comment by lmarceau** [comment]
@junbio thanks for the fix! I didn't realized when I first open ticket #9806 but the Share option also crashes when we long press on a pocket tile. Would you mind also making the fix for it? It seems that you'll also need to unwrap the pocketCell like it was done for the topSiteCell (so same fix).

We can also open another ticket for it if you wish.

File: Client/Frontend/Home/FirefoxHomeViewController.swift [hunk 1]
```diff
@@ -1255,8 +1255,8 @@ extension FirefoxHomeViewController: HomePanelContextMenu {
 
         switch Section(indexPath.section) {
         case .topSites:
-            if let topSiteCell = self.collectionView?.cellForItem(at: IndexPath(row: 0, section: 0)) as? ASHorizontalScrollCell {
-                sourceView = topSiteCell.collectionView.cellForItem(at: indexPath)
+            if let topSiteCell = self.collectionView?.cellForItem(at: IndexPath(row: 0, section: indexPath.section)) as? ASHorizontalScrollCell {
```
Comment (lmarceau) [inline Client/Frontend/Home/FirefoxHomeViewController.swift:1258]: nit: While we're at it, we can remove the `self.` here on that collectionView

**Comment by junbio** [comment]
Made the changes as requested. Thanks!

**Review by lmarceau — APPROVED** [review]
Thanks @junbio 😄

**Pull request merged**